### PR TITLE
1884761: Candlepin presents a different number of certificates for a consumer based on accept header (ENT-3127)

### PR DESCRIPTION
**Pull request by wolfdale**
Changes  :-
- Changes made to export content access certs for a consumer who's owner in SCA mode.
- Added Spec test to verify it.

### CODE REVIEW

**Review by Januson — CHANGES_REQUESTED** [review]
Changes look good. I have a few suggestions regarding the exporter and its tests.

File: server/src/main/java/org/candlepin/sync/Exporter.java [hunk 1]
```diff
@@ -197,6 +206,7 @@ public File getEntitlementExport(Consumer consumer, Set<Long> serials) throws Ex
 
             exportMeta(baseDir, null);
             exportEntitlementsCerts(baseDir, consumer, serials, false);
+            exportContentAccessCerts(baseDir, consumer);
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/sync/Exporter.java:204]: Added a unit test for it.

File: server/src/main/java/org/candlepin/sync/Exporter.java [hunk 2]
```diff
@@ -120,7 +125,9 @@ public Exporter(ConsumerTypeCurator consumerTypeCurator, OwnerCurator ownerCurat
         CdnExporter cdnExporter,
         ProductCurator productCurator,
         SyncUtils syncUtils, ExportExtensionAdapter extensionAdapter,
-        ModelTranslator translator) {
+        ModelTranslator translator,
+        ContentAccessManager contentAccessManager,
+        ContentAccessCertExporter contentAccessCertExporter) {
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/sync/Exporter.java]: Yes agreed.

File: server/src/main/java/org/candlepin/sync/ContentAccessCertExporter.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * in this software or its documentation.
+ */
+package org.candlepin.sync;
+
+import org.candlepin.model.ContentAccessCertificate;
+
+import java.io.IOException;
+import java.io.Writer;
+
+public class ContentAccessCertExporter {
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/sync/ContentAccessCertExporter.java]: Sure, moved all the writer logic inside new certificate exporter class.

File: server/src/main/java/org/candlepin/sync/Exporter.java [hunk 4]
```diff
@@ -423,6 +433,48 @@ private void exportEntitlementsCerts(File baseDir, Consumer consumer,
...
+            if (contentAccessCert != null) {
+                File contentAccessCertDir = new File(baseDir.getCanonicalPath(),
+                    "content_access_certificates");
+                contentAccessCertDir.mkdir();
+                log.debug("Exporting content access certificate: " + contentAccessCert.getSerial());
+                File file = new File(contentAccessCertDir.getCanonicalPath(),
+                    contentAccessCert.getSerial().getId() + ".pem");
+                FileWriter writer = null;
+
+                try {
```
[the remark this answers is not in the document]
Comment (wolfdale) [reply server/src/main/java/org/candlepin/sync/Exporter.java]: Changed.

**Review by Januson — APPROVED** [review]

**Pull request merged**